packages/ddl-shim/src/dermatology.ts

```diff
@@ -0,0 +1,220 @@
+import * as DDL2 from "./ddl2";
+
+const classMappings: any = {
+    c3chart_Bar: "chart_Bar",
+    c3chart_Column: "chart_Column",
+    c3chart_Pie: "chart_Pie",
+    c3chart_Area: "chart_Area",
+    c3chart_Line: "chart_Line",
+    amchart_Bar: "chart_Bar",
+    amchart_Column: "chart_Column",
+    amchart_Pie: "chart_Pie",
+    amchart_Area: "chart_Area",
+    amchart_Line: "chart_Line",
+    google_Bar: "chart_Bar",
+    google_Column: "chart_Column",
+    google_Pie: "chart_Pie",
+    google_Area: "chart_Area",
+    google_Line: "chart_Line",
+    other_Table: "dgrid_Table"
+};
+const propertyMappings: any = {
+    xAxisLabelRotation: [
+        { name: "xAxisOverlapMode", transform: (n: any) => "rotate" },
+        { name: "xAxisLabelRotation", transform: (n: any) => n }
+    ],
+    tooltipLabelColor: {
+        name: "tooltipLabelColor"
+    },
+    tooltipSeriesColor: {
+        name: "tooltipSeriesColor"
+    },
+    tooltipValueColor: {
+        name: "tooltipValueColor"
+    },
+    tooltipValueFormat: {
+        name: "tooltipValueFormat"
+    },
+    timePattern: {
+        name: "xAxisTypeTimePattern"
+    },
+    smoothLines: {
+        name: "interpolate",
+        transform: (n: any) => {
+            if (n === false) return "linear";
+            return "catmullRom";
+        }
+    },
+    holePercent: {
+        name: "innerRadius"
+    },
+    flip: {
+        name: "orientation",
+        transform: (n: any) => n ? "vertical" : "horizontal"
+    },
+    bottomText: {
+        name: "xAxisTitle"
+    },
+    xAxisTypeTimePattern: {
+        name: "xAxisTypeTimePattern"
+    },
+    yAxisTypeTimePattern: {
+        name: "yAxisTypeTimePattern"
+    },
+    valueFormat: {
+        name: "tooltipValueFormat"
+    },
+    stacked: {
+        name: "yAxisStacked"
+    },
+    showYGrid: {
+        name: "yAxisGuideLines"
+    },
+    showXGrid: {
+        name: "xAxisGuideLines"
+    },
+    showValueLabel: {
+        name: "showValue"
+    },
+    low: {
+        name: "yAxisDomainLow"
+    },
+    high: {
+        name: "yAxisDomainHigh"
+    },
+    fillOpacity: {
+        name: "interpolateFillOpacity"
+    },
+    areaFillOpacity: {
+        name: "interpolateFillOpacity"
+    },
+    showToolbar: {
+        name: "titleVisible"
+    },
+    showCSV: {
+        name: "downloadButtonVisible"
+    }
+};
+
+function findKeyVal(object: any, key: any, val: any) {
+    let value;
+
+    Object.keys(object).some(k => {
```

Comment:
loop not needed (I think)?

Reply:
The loop is needed because the objects aren't index by their id, or ordered in a predictable way.